src/main/java/io/confluent/connect/hdfs/orc/OrcHiveUtil.java

```diff
@@ -0,0 +1,105 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.connect.hdfs.orc;
+
+import io.confluent.connect.hdfs.HdfsSinkConnectorConfig;
+import io.confluent.connect.hdfs.hive.HiveMetaStore;
+import io.confluent.connect.hdfs.hive.HiveUtil;
+import io.confluent.connect.hdfs.partitioner.Partitioner;
+import io.confluent.connect.storage.errors.HiveMetaStoreException;
+import io.confluent.connect.storage.hive.HiveSchemaConverter;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.metastore.TableType;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.ql.io.orc.OrcInputFormat;
+import org.apache.hadoop.hive.ql.io.orc.OrcOutputFormat;
+import org.apache.hadoop.hive.ql.io.orc.OrcSerde;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.kafka.connect.data.Schema;
+
+import java.util.List;
+
+public class OrcHiveUtil extends HiveUtil {
+  private final HdfsSinkConnectorConfig config;
+
+  public OrcHiveUtil(HdfsSinkConnectorConfig config, HiveMetaStore hiveMetaStore) {
+    super(config, hiveMetaStore);
+    this.config = config;
+  }
+
+  @Override
+  public void alterSchema(String database, String tableName, Schema schema) {
+    Table table = hiveMetaStore.getTable(database, tableName);
+    List<FieldSchema> columns = HiveSchemaConverter.convertSchema(schema);
+    table.setFields(columns);
+    hiveMetaStore.alterTable(table);
+  }
+
+  @Override
+  public void createTable(
+      String database,
+      String tableName,
+      Schema schema,
+      Partitioner partitioner
+  ) throws HiveMetaStoreException {
+    Table table = constructOrcTable(database, tableName, schema, partitioner);
+    hiveMetaStore.createTable(table);
+  }
+
+  private Table constructOrcTable(
+      String database,
+      String tableName,
+      Schema schema,
+      Partitioner partitioner
+  ) throws HiveMetaStoreException {
+
+    Table table = newTable(database, tableName);
+    table.setTableType(TableType.EXTERNAL_TABLE);
+    table.getParameters().put("EXTERNAL", "TRUE");
+
+    // tableName is always the topic name
+    String tablePath = hiveDirectoryName(url, config.getTopicsDirFromTopic(tableName), tableName);
+    table.setDataLocation(new Path(tablePath));
+    table.setSerializationLib(getHiveOrcSerde());
+
+    try {
+      table.setInputFormatClass(getHiveOrcInputFormat());
+      table.setOutputFormatClass(getHiveOrcOutputFormat());
+    } catch (HiveException e) {
+      throw new HiveMetaStoreException("Cannot find input/output format:", e);
+    }
+
+    // convert copycat schema schema to Hive columns
```

Comment:
haha is this a stale comment from one of the other HiveUtils?
```suggestion
    // convert connect schema to Hive columns
```

Reply:
indeed it is :P time to purge the comments